sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -254,13 +254,15 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case i @ InsertIntoTable(UnresolvedCatalogRelation(tableMeta), _, _, _, _)
-        if DDLUtils.isDatasourceTable(tableMeta) =>
+        if DDLUtils.isDatasourceTable(tableMeta) &&
+          DDLUtils.convertSchema(tableMeta, sparkSession) =>
```

Comment:
I do not think this is a right fix. If the original table is the native data source table, we will always use our parquet/orc reader instead of hive serde.

Reply:
do you mean any table built through df.write.format("..") should be taken as a data source table no matter we register it with HMS or not

Reply:
If you are using `format("parquet")` to create a new table, it will be a data source table. We always use the native reader/writer to read/write such a table.

Reply:
ok